manager.go

```diff
@@ -1823,11 +1825,8 @@ func (m *Manager) newCachedNetlinkSocket(nsHandle uint64, nsID uint32) (*Netlink
 
 // GetNetlinkSocket - Returns a netlink socket in the requested network namespace from cache or creates a new one.
 func (m *Manager) GetNetlinkSocket(nsHandle uint64, nsID uint32) (*NetlinkSocket, error) {
-	m.stateLock.Lock()
-	defer m.stateLock.Unlock()
-	if m.state < initialized {
-		return nil, ErrManagerNotInitialized
-	}
+	m.nscLock.Lock()
+	defer m.nscLock.Unlock()
```

Comment:
Any reason to have these mutexes outside the function, rather than part of `getNetlinkSocket`?

Reply:
It's because other manager functions are calling the `getNetlinkSocket` function while already having the mutex locked

Reply:
They have the `stateLock` mutext locked correct? Not the `nscLock` mutex?

Reply:
Oh you are absolutely right

Reply:
Should be good now